pymc_marketing/mmm/fourier.py

```diff
@@ -261,63 +259,53 @@ def generate_fourier_modes(
     )
 
 
-class FourierBase:
+class FourierBase(BaseModel):
     """Base class for Fourier seasonality transformations.
 
     Parameters
     ----------
     n_order : int
         Number of fourier modes to use.
+    days_in_period : float
+        Number of days in a period.
     prefix : str, optional
         Alternative prefix for the fourier seasonality, by default None or
         "fourier"
     prior : Prior, optional
         Prior distribution for the fourier seasonality beta parameters, by
-        default None
+        default `Prior("Laplace", mu=0, b=1)`
     name : str, optional
         Name of the variable that multiplies the fourier modes, by default None
-
-    Attributes
-    ----------
-    days_in_period : float
-        Number of days in a period.
-    prefix : str
-        Name of model coordinates
-    default_prior : Prior
-        Default prior distribution for the fourier seasonality
-        beta parameters.
+    variable_name : str, optional
+        Name of the variable that multiplies the fourier modes, by default None
 
     """
 
-    days_in_period: float
-    prefix: str = "fourier"
+    n_order: int = Field(..., gt=0)
+    days_in_period: float = Field(..., gt=0)
+    prefix: str | None = Field("fourier")
+    prior: InstanceOf[Prior] = Field(Prior("Laplace", mu=0, b=1))
+    name: str | None = Field(None)
+    variable_name: str | None = Field("None")
```

Comment:
Think we should consolidate these? `name` is only used as `variable_name` if non None

Reply:
Yes! Solved in https://github.com/pymc-labs/pymc-marketing/pull/809/commits/36ef9f3d613fd8949d34b496aa7bc4b8c38f1773